cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsPublicKeyConnector.java

```diff
@@ -83,6 +85,21 @@ public boolean exists(PublicKeyDescribeRequest request) {
         return false;
     }
 
+    @Override
+    public String rawPublicKey(PublicKeyDescribeRequest request) {
+        LOGGER.debug("Describe public key {} in {} region on AWS", request.getPublicKeyId(), request.getRegion());
+        AwsCredentialView awsCredential = new AwsCredentialView(request.getCredential());
+        try {
+            AmazonEc2Client client = awsClient.createEc2Client(awsCredential, request.getRegion());
+            return rawPublicKey(client, request.getPublicKeyId());
+        } catch (Exception e) {
+            String errorMessage = String.format("Failed to describe public key [%s:'%s', region: '%s'], detailed message: %s",
+                    getType(awsCredential), getAwsId(awsCredential), request.getRegion(), e.getMessage());
+            LOGGER.error(errorMessage, e);
+        }
+        return null;
```

Comment:
could be moved to the catch part, so it would be quickly obvious that this should happen only when there is an exception